crates/fee/src/tests.rs

```diff
@@ -13,3 +16,159 @@ fn should_get_issue_fee() {
+        assert_ok!(Fee::set_refund_fee(Origin::root(), fee));
+        assert_eq!(<RefundFee<Test>>::get(), fee);
+    })
+}
+
+#[test]
+fn should_set_premium_redeem_fee() {
+    run_test(|| {
+        let large_fee = UnsignedFixedPoint::checked_from_rational::<u128, u128>(101, 100).unwrap(); // 101%
+        assert_noop!(
+            Fee::set_premium_redeem_fee(Origin::root(), large_fee),
+            TestError::AboveMaxExpectedValue
+        );
+
+        let fee = UnsignedFixedPoint::checked_from_rational(100, 100).unwrap(); // 100%
+        assert_noop!(
+            Fee::set_premium_redeem_fee(Origin::signed(6), fee),
+            DispatchError::BadOrigin
+        );
+        assert_ok!(Fee::set_premium_redeem_fee(Origin::root(), fee));
+        assert_eq!(<PremiumRedeemFee<Test>>::get(), fee);
+    })
+}
+
+#[test]
+fn should_set_punishment_fee() {
+    run_test(|| {
+        let large_fee = UnsignedFixedPoint::checked_from_rational::<u128, u128>(101, 100).unwrap(); // 101%
+        assert_noop!(
+            Fee::set_punishment_fee(Origin::root(), large_fee),
+            TestError::AboveMaxExpectedValue
+        );
+
+        let fee = UnsignedFixedPoint::checked_from_rational(100, 100).unwrap(); // 100%
+        assert_noop!(
+            Fee::set_punishment_fee(Origin::signed(6), fee),
+            DispatchError::BadOrigin
+        );
+        assert_ok!(Fee::set_punishment_fee(Origin::root(), fee));
+        assert_eq!(<PunishmentFee<Test>>::get(), fee);
+    })
+}
+
+#[test]
+fn should_set_replace_griefing_collateral() {
+    run_test(|| {
+        let large_griefing_collateral = UnsignedFixedPoint::checked_from_rational::<u128, u128>(101, 100).unwrap(); // 10100%
+        assert_noop!(
+            Fee::set_replace_griefing_collateral(Origin::root(), large_griefing_collateral),
+            TestError::AboveMaxExpectedValue
+        );
+
+        let griefing_collateral = UnsignedFixedPoint::checked_from_rational(100, 100).unwrap(); // 100%
+        assert_noop!(
+            Fee::set_replace_griefing_collateral(Origin::signed(6), griefing_collateral),
+            DispatchError::BadOrigin
+        );
+        assert_ok!(Fee::set_replace_griefing_collateral(
+            Origin::root(),
+            griefing_collateral
+        ));
+        assert_eq!(<ReplaceGriefingCollateral<Test>>::get(), griefing_collateral);
+    })
+}
+
+#[test]
+fn should_set_theft_fee() {
+    run_test(|| {
+        let large_fee = UnsignedFixedPoint::checked_from_rational::<u128, u128>(101, 100).unwrap(); // 101%
+        assert_noop!(
+            Fee::set_theft_fee(Origin::root(), large_fee),
+            TestError::AboveMaxExpectedValue
+        );
+
+        let fee = UnsignedFixedPoint::checked_from_rational(100, 100).unwrap(); // 100%
+        assert_noop!(Fee::set_theft_fee(Origin::signed(6), fee), DispatchError::BadOrigin);
+        assert_ok!(Fee::set_theft_fee(Origin::root(), fee));
+        assert_eq!(<TheftFee<Test>>::get(), fee);
+    })
+}
```

Comment:
I think we can cut down the amount of code we need to maintain, perhaps you can define a macro which generates test cases for each storage field?

Reply:
Might even be able to use a function rather than a macro - one that takes a `impl Fn` and whatever type `<TheftFee<Test>>` is.